tinygrad/codegen/linearizer.py

```diff
@@ -314,10 +314,7 @@ def linearize(self):
         fake_global_idxs = [x*0 for x in global_idxs]
         self.global_store(-1, fake_global_idxs+local_idxs+fake_reduce_idxs+upcast_idxs, acc)  # store accumulators
         self.uop(UOps.BARRIER, None, [], ())
-        self.uop(UOps.ENDLOOP, None, [], (loop_local_idxs, "local"))
-
-        # local indexs are over, 0 them out
-        local_idxs = [x*0 for x in local_idxs]
+        self.uop(UOps.ENDLOOP, None, [], ([loop_local_idxs[i]*0 if i < len(self.group_for_reduce) else loop_local_idxs[i] for i in range(0,self.local_dims+len(self.group_for_reduce))], "local"))
```

Comment:
Seems you also close the whole local loop here (not only group_for_reduce part. [`global+local`](https://github.com/flammit/tinygrad/commit/109fec79d83ba22f185a8a3ff3f140ecb080022e#diff-63d57daaca59e5495bbb108da1c13e782f4a408db8503e015e8094ac07c87ea6R357) should close all other locals?

Reply:
I believe ENDLOOP out with two elements works.  See above for why it's only the group_for_reduce part.